prometheus/promhttp/http_test.go

```diff
@@ -331,3 +331,162 @@ func TestHandlerTimeout(t *testing.T) {
 
 	close(c.Block) // To not leak a goroutine.
 }
+
+func TestGetWriter(t *testing.T) {
```

Comment:
This is good, but I believe we need to improve bigger tests to at least use gzip on top of this. If we would merge this code it would break ppl I think (wrong defer scope) 🙈

Reply:
I added more tests that test the whole compression with gzip and zstd, let me know if that is sufficient.